phoenix-core/src/main/java/org/apache/phoenix/compile/OrderByCompiler.java

```diff
@@ -192,6 +192,15 @@ public static OrderBy compile(StatementContext context,
             }
             compiler.reset();
         }
+
+        //If we are not ordered we shouldn't be using RVC Offset
+        //I think this makes sense for the pagination case but perhaps we can relax this for
+        //other use cases.
+        //Note If the table is salted we still mark as row ordered in this code path
+        if(offset.getByteOffset().isPresent() && orderByExpressions.isEmpty()){
+            throw new SQLException("Do not allow non ORDER BY with RVC OFFSET");
```

Comment:
hmm I don't understand why we disallow this. The simple case of using OFFSET can be done without using `ORDER BY` right? Besides, if this is already inside `OrderByCompiler#compile`, shouldn't there already be a check for `!orderByExpressions.isEmpty()`? If not, I don't understand how it can/should ever reach this code path in the first place.

Reply:
Order by is empty is not necessarily needed if they specify a row order or get one from rowkey order property we should let the query through.

Reply:
Will change error message as discussed.

Reply:
@dbwong ping on this change